src/main/java/hudson/plugins/git/extensions/impl/SparseCheckoutPath.java

```diff
@@ -56,20 +59,41 @@ public String toString() {
         return path;
     }
 
-    private static class SparseCheckoutPathToPath implements Function<SparseCheckoutPath, String>, Serializable {
-        public String apply(@NonNull SparseCheckoutPath sparseCheckoutPath) {
-            return sparseCheckoutPath.getPath();
-        }
+    public Descriptor<SparseCheckoutPath> getDescriptor() {
+        return Jenkins.get().getDescriptor(getClass());
     }
 
-    public Descriptor<SparseCheckoutPath> getDescriptor()
-    {
-        return Jenkins.get().getDescriptor(getClass());
+    public static class SparseCheckoutPathToPath implements Function<SparseCheckoutPath, String>, Serializable {
+        @Nullable
+        private EnvVars envVars;
+
+        SparseCheckoutPathToPath(@Nullable EnvVars envVars) {
+            this.envVars = envVars;
+        }
+
+        public String apply(@NonNull SparseCheckoutPath sparseCheckoutPath) {
+            String path = sparseCheckoutPath.getPath();
+            if (envVars == null) {
+                return path;
+            }
+
+            // Pattern to look for substring of the form ${ENV_VAR}.
+            Pattern pattern = Pattern.compile("\\$\\{(.*?)\\}");
+            Matcher matcher = pattern.matcher(path);
+            while (matcher.find()) {
+                String varName = matcher.group(1);
+                String value = envVars.get(varName, "");
+                path = path.replace("${" + varName + "}", value);
+            }
+            return path;
+        }
     }
 
     @Extension
     public static class DescriptorImpl extends Descriptor<SparseCheckoutPath> {
         @Override
-        public String getDisplayName() { return "Path"; }
```

Comment:
Please don't make white space only changes in the git plugin.  See the [contributing file](https://github.com/jenkinsci/git-plugin/blob/master/CONTRIBUTING.adoc#contributing-to-the-git-plugin) for more details.

Reply:
As mentioned this was not intentional either.